ta/os_test/os_test.c

```diff
@@ -758,20 +760,112 @@ static bool optee_pager_with_small_pool(void)
 	return rc;
 }
 
+struct consume_elem {
+	SLIST_ENTRY(consume_elem) link;
+};
+SLIST_HEAD(consume_elem_head, consume_elem);
+
+static void consume_all_heap(struct consume_elem_head *h)
```

Comment:
Function does what it should do, but if 4096 isn't an upper limit and we just want to make sure to consume all heap, then an alternative and likely faster approach could be to start doubling the size until we get an error, and then after that half the size like you've done. I believe this could save a few loops and keep the linked list shorter. But, the question is whether it's worth doing something like that here? Perhaps just add an  `else { p_sz *= 2 }` to the innermost loop 
and take care of  `SLIST_INSERT_HEAD` accordingly would achieve this?